### PR TITLE
[ios_platform_images] remove deprecated APIs

### PR DESCRIPTION
**Pull request by jonahwilliams**
These APIs were deprecated before the last stable release, and will be removed in a future release of Flutter

### CODE REVIEW

**Comment by Hixie** [comment]
test-exempt: code refactor with no semantic change

File: packages/ios_platform_images/pubspec.yaml [hunk 1]
```diff
@@ -2,11 +2,11 @@ name: ios_platform_images
 description: A plugin to share images between Flutter and iOS in add-to-app setups.
 repository: https://github.com/flutter/plugins/tree/main/packages/ios_platform_images
 issue_tracker: https://github.com/flutter/flutter/issues?q=is%3Aissue+is%3Aopen+label%3A%22p%3A+ios_platform_images%22
-version: 0.2.0+9
+version: 0.3.0
 
 environment:
   sdk: ">=2.14.0 <3.0.0"
-  flutter: ">=2.10.0"
+  flutter: ">=3.3.0"
```
Comment (jmagman) [inline packages/ios_platform_images/pubspec.yaml:9]: @stuartmorgan said upping to 3.3 was okay per trailing stable policies in https://github.com/flutter/flutter/issues/101965#issuecomment-1232771019.

**Review by jmagman — APPROVED** [review]
The deprecation details are at https://docs.flutter.dev/release/breaking-changes/image-provider-load-buffer.

LGTM

File: packages/ios_platform_images/pubspec.yaml (hunk 1 above)
Comment (stuartmorgan) [reply packages/ios_platform_images/pubspec.yaml:9]: Yep, this is also documented at https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#supported-flutter-versions

**Review by stuartmorgan — APPROVED** [review]
LGTM!

**Comment by stuartmorgan** [comment]
There was a scheduling issue with Cirrus macOS machines; trying a re-run now in the hopes that it's cleared up.

**Comment by jonahwilliams** [comment]
How do I fix this?

**Comment by stuartmorgan** [comment]
Try merging the latest `main` in again to re-trigger tests. AFAIK there's no way to manually re-run `ci.yaml validation`, unfortunately.

**Pull request merged**